src/cheri_insts.sail

```diff
@@ -661,16 +661,16 @@ function clause execute (CTestSubset(rd, cs1, cs2)) = {
   let cs1_val = C(cs1);
   let cs2_val = C(cs2);
 
-  let (cs2_base, cs2_top) = getCapBounds(cs2_val);
```

Comment:
Why not keep the API of `getCapBounds` the same and have it return `base+length` rather than make changes here? It's not a big deal just a matter of taste and what changes are essential to this PR vs. incidental. I think if you change the meaning of the function you should change its name to avoid hidden problems with later merges. Reasons to prefer top rather than length are: 
1) it is usually what you want as evidenced by the fact that every one of these changes includes also substituting top for base + length
2) it means you only have to think about the potential for overflow in one place